File: CS/SpreadsheetDocServerPivotAPI/CodeUtils/SpreadsheetCodeResultViewer.cs
using Microsoft.CSharp;
using System.CodeDom.Compiler;

namespace SpreadsheetDocServerPivotAPI
{
    public abstract class SpreadsheetExampleCodeEvaluator : ExampleCodeEvaluator
    {
        protected override string GetModuleAssembly()
        {
            return AssemblyInfo.SRAssemblySpreadsheetCore;
        }
        protected override string GetExampleClassName()
        {
            return "SpreadsheetCodeResultViewer.ExampleItem";
        }
    }
    #region SpreadsheetCSExampleCodeEvaluator
    public partial class SpreadsheetCSExampleCodeEvaluator : SpreadsheetExampleCodeEvaluator
    {

        protected override CodeDomProvider GetCodeDomProvider()
        {
            return new CSharpCodeProvider();
        }

        const string codeStart =
      "using System;\r\n" +
      "using DevExpress.Spreadsheet;\r\n" +
      "using DevExpress.Spreadsheet.Charts;\r\n" +
      "using DevExpress.Spreadsheet.Drawings;\r\n" +
      "using DevExpress.XtraSpreadsheet.Export;\r\n" +
      "using System.Drawing;\r\n" +
      "using System.Windows.Forms;\r\n" +
      "using DevExpress.XtraPrinting;\r\n" +
      "using DevExpress.XtraPrinting.Control;\r\n" +
      "using DevExpress.Utils;\r\n" +
      "using System.IO;\r\n" +
      "using System.Diagnostics;\r\n" +
      "using System.Xml;\r\n" +
      "using System.Data;\r\n" +
      "using System.Collections.Generic;\r\n" +
      "using System.Globalization;\r\n" +
      "using Formatting = DevExpress.Spreadsheet.Formatting;\r\n" +
      "namespace SpreadsheetCodeResultViewer { \r\n" +
      "public class ExampleItem { \r\n" +
      "        public static void Process(Workbook workbook) { \r\n" +
      "\r\n";

        const string codeEnd =
        "       \r\n }\r\n" +
        "    }\r\n" +
        "}\r\n";
        protected override string CodeStart { get { return codeStart; } }
        protected override string CodeEnd { get { return codeEnd; } }
    }
    #endregion
    #region SpreadsheetVbExampleCodeEvaluator
    public partial class SpreadsheetVbExampleCodeEvaluator : SpreadsheetExampleCodeEvaluator
    {

        protected override CodeDomProvider GetCodeDomProvider()
        {
            return new Microsoft.VisualBasic.VBCodeProvider();
        }
        const string codeStart =
      "Imports Microsoft.VisualBasic\r\n" +
      "Imports System\r\n" +
      "Imports DevExpress.Spreadsheet\r\n" +
      "Imports DevExpress.Spreadsheet.Charts\r\n" +
      "Imports DevExpress.Spreadsheet.Drawings\r\n" +
      "Imports DevExpress.XtraSpreadsheet.Export\r\n" +
      "Imports System.Drawing\r\n" +
      "Imports System.Windows.Forms\r\n" +
      "Imports DevExpress.XtraPrinting\r\n" +
      "Imports DevExpress.XtraPrinting.Control\r\n" +
      "Imports DevExpress.Utils\r\n" +
      "Imports System.IO\r\n" +
      "Imports System.Diagnostics\r\n" +
      "Imports System.Xml\r\n" +
      "Imports System.Data\r\n" +
      "Imports System.Collections.Generic\r\n" +
      "Imports System.Globalization\r\n" +
      "Imports Formatting = DevExpress.Spreadsheet.Formatting\r\n" +
      "Namespace SpreadsheetCodeResultViewer\r\n" +
      "	Public Class ExampleItem\r\n" +
      "		Public Shared Sub Process(ByVal workbook As Workbook)\r\n" +
      "\r\n";

        const string codeEnd =
        "\r\n		End Sub\r\n" +
        "	End Class\r\n" +
        "End Namespace\r\n";

        protected override string CodeStart { get { return codeStart; } }
        protected override string CodeEnd { get { return codeEnd; } }
    }
    #endregion
}

File: VB/SpreadsheetDocServerPivotAPI/CodeUtils/SpreadsheetCodeResultViewer.vb
Imports Microsoft.CSharp
Imports System.CodeDom.Compiler

Namespace SpreadsheetDocServerPivotAPI
    Public MustInherit Class SpreadsheetExampleCodeEvaluator
        Inherits ExampleCodeEvaluator

        Protected Overrides Function GetModuleAssembly() As String
            Return AssemblyInfo.SRAssemblySpreadsheetCore
        End Function
        Protected Overrides Function GetExampleClassName() As String
            Return "SpreadsheetCodeResultViewer.ExampleItem"
        End Function
    End Class
    #Region "SpreadsheetCSExampleCodeEvaluator"
    Partial Public Class SpreadsheetCSExampleCodeEvaluator
        Inherits SpreadsheetExampleCodeEvaluator

        Protected Overrides Function GetCodeDomProvider() As CodeDomProvider
            Return New CSharpCodeProvider()
        End Function


        Private Const codeStart_Renamed As String = "using System;" & ControlChars.CrLf & "using DevExpress.Spreadsheet;" & ControlChars.CrLf & "using DevExpress.Spreadsheet.Charts;" & ControlChars.CrLf & "using DevExpress.Spreadsheet.Drawings;" & ControlChars.CrLf & "using DevExpress.XtraSpreadsheet.Export;" & ControlChars.CrLf & "using System.Drawing;" & ControlChars.CrLf & "using System.Windows.Forms;" & ControlChars.CrLf & "using DevExpress.XtraPrinting;" & ControlChars.CrLf & "using DevExpress.XtraPrinting.Control;" & ControlChars.CrLf & "using DevExpress.Utils;" & ControlChars.CrLf & "using System.IO;" & ControlChars.CrLf & "using System.Diagnostics;" & ControlChars.CrLf & "using System.Xml;" & ControlChars.CrLf & "using System.Data;" & ControlChars.CrLf & "using System.Collections.Generic;" & ControlChars.CrLf & "using System.Globalization;" & ControlChars.CrLf & "using Formatting = DevExpress.Spreadsheet.Formatting;" & ControlChars.CrLf & "namespace SpreadsheetCodeResultViewer { " & ControlChars.CrLf & "public class ExampleItem { " & ControlChars.CrLf & "        public static void Process(Workbook workbook) { " & ControlChars.CrLf & ControlChars.CrLf


        Private Const codeEnd_Renamed As String = "       " & ControlChars.CrLf & " }" & ControlChars.CrLf & "    }" & ControlChars.CrLf & "}" & ControlChars.CrLf
        Protected Overrides ReadOnly Property CodeStart() As String
            Get
                Return codeStart_Renamed
            End Get
        End Property
        Protected Overrides ReadOnly Property CodeEnd() As String
            Get
                Return codeEnd_Renamed
            End Get
        End Property
    End Class
    #End Region
    #Region "SpreadsheetVbExampleCodeEvaluator"
    Partial Public Class SpreadsheetVbExampleCodeEvaluator
        Inherits SpreadsheetExampleCodeEvaluator

        Protected Overrides Function GetCodeDomProvider() As CodeDomProvider
            Return New Microsoft.VisualBasic.VBCodeProvider()
        End Function

        Private Const codeStart_Renamed As String = "Imports Microsoft.VisualBasic" & ControlChars.CrLf & "Imports System" & ControlChars.CrLf & "Imports DevExpress.Spreadsheet" & ControlChars.CrLf & "Imports DevExpress.Spreadsheet.Charts" & ControlChars.CrLf & "Imports DevExpress.Spreadsheet.Drawings" & ControlChars.CrLf & "Imports DevExpress.XtraSpreadsheet.Export" & ControlChars.CrLf & "Imports System.Drawing" & ControlChars.CrLf & "Imports System.Windows.Forms" & ControlChars.CrLf & "Imports DevExpress.XtraPrinting" & ControlChars.CrLf & "Imports DevExpress.XtraPrinting.Control" & ControlChars.CrLf & "Imports DevExpress.Utils" & ControlChars.CrLf & "Imports System.IO" & ControlChars.CrLf & "Imports System.Diagnostics" & ControlChars.CrLf & "Imports System.Xml" & ControlChars.CrLf & "Imports System.Data" & ControlChars.CrLf & "Imports System.Collections.Generic" & ControlChars.CrLf & "Imports System.Globalization" & ControlChars.CrLf & "Imports Formatting = DevExpress.Spreadsheet.Formatting" & ControlChars.CrLf & "Namespace SpreadsheetCodeResultViewer" & ControlChars.CrLf & "	Public Class ExampleItem" & ControlChars.CrLf & "		Public Shared Sub Process(ByVal workbook As Workbook)" & ControlChars.CrLf & ControlChars.CrLf


        Private Const codeEnd_Renamed As String = ControlChars.CrLf & "		End Sub" & ControlChars.CrLf & "	End Class" & ControlChars.CrLf & "End Namespace" & ControlChars.CrLf

        Protected Overrides ReadOnly Property CodeStart() As String
            Get
                Return codeStart_Renamed
            End Get
        End Property
        Protected Overrides ReadOnly Property CodeEnd() As String
            Get
                Return codeEnd_Renamed
            End Get
        End Property
    End Class
    #End Region
End Namespace
